task_3/module/ec2/main.tf

```diff
@@ -0,0 +1,75 @@
+# DATASOURCES
+
+data "aws_ami" "ami" {
+  most_recent = var.ami_param.most_recent
+  name_regex  = var.ami_param.name_regex
+  owners      = var.ami_param.owners
+}
+
+data "aws_vpc" "vpc" {
```

Comment:
I would recommend you to move all network related data_sources to root module. This will prevent issues and improve code reuse. 
You also can move ami related data source outside the ec2 module. This will give your module a better reuse.

Reply:
subnet definition section moved to root tf manifest. ec2 module adaptated
#[task_3/main.tf](data source section)